study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/EventBuilder.java

```diff
@@ -79,9 +89,45 @@ private void insertEvents(Handle handle) {
         }
     }
 
+    private void insertLabeledEvents(Handle handle, String[] eventLabelArray) {
+        if (!cfg.hasPath("events")) {
+            return;
+        }
+        Set<String> eventLabelsToLoad = new HashSet<>(Arrays.asList(eventLabelArray));
+        List<Config> allLabeledCfgs = cfg.getConfigList("events").stream()
+                .filter(eventCfg -> eventCfg.hasPath("label") && eventCfg.getString("label") != null)
+                .collect(toList());
+       List<String> allCfgLabels = allLabeledCfgs.stream().map(cfg -> cfg.getString("label")).collect(toList());
+       Set<String> notFoundEventLabels = eventLabelsToLoad.stream().filter(eventName -> !allCfgLabels.contains(eventName)).collect(toSet());
+       if (!notFoundEventLabels.isEmpty()) {
+           throw new DDPException("Could not find events " + String.join(", ", notFoundEventLabels));
+       }
+       if (allCfgLabels.size() > eventLabelsToLoad.size()) {
+           throw new DDPException("Found duplicate names in event configuration entries");
+       }
+       List<Config> eventCfgsToLoad = allLabeledCfgs.stream()
+                   .filter(cfg -> allCfgLabels.contains(cfg.getString("label")))
```

Comment:
Did we mean to check against the `eventLabelsToLoad` set?

Reply:
Wrong too.
dbea8bc3b184e2e39ed61bc53e39c4bef2aa58bb